### PR TITLE
httpsec: externalize ClientIP()

### PR DESCRIPTION
**Pull request by Julio-Guerra**
Externalise dd-trace-go's `httpsec.ClientIP()` in this repo to be able to reuse it in the agent. To do so:
1. Copy/Paste the original implementation.
2. Let dd-trace-go configure the function by always passing the list of HTTP headers to monitor.
3. Found a mistake fixed with multiple headers where the `break` statement was breaking the inner loop rather than the main one. This was preventing stopping the header's loop as soon as a global IP was found.

### CODE REVIEW

**Review by Hellzy — COMMENTED** [review]
I feel like this code has become cleaner over time, it's nice to see 😃
Couple comments on my side.

File: httpsec/client_ip_test.go [hunk 1]
```diff
@@ -0,0 +1,350 @@
...
+	//			headers:    map[string]string{header: ipv6Private},
+	//			expectedIP: netip.MustParseAddr(ipv6Global),
+	//		},
+	//		ipTestCase{
+	//			name:       "ipv6-global-remoteaddr-global-ip-header." + header,
+	//			remoteAddr: ipv4Global,
+	//			headers:    map[string]string{header: ipv6Global},
+	//			expectedIP: netip.MustParseAddr(ipv6Global),
+	//		})
+	//}
```
[the remark this answers is not in the document]
Comment (Julio-Guerra) [reply httpsec/client_ip_test.go]: Restored in [4e9b83a](https://github.com/DataDog/appsec-internal-go/pull/5/commits/4e9b83a93282aae408e47fd27f8bf996eefb0e0a)

File: httpsec/client_ip.go [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+		tags["http.client_ip"] = clientIP.String()
+	}
+	return tags
+}
+
+// ClientIP returns the first public IP address found in the given headers. If
+// none is present, it returns the first valid IP address present, possibly
+// being a local IP address. The remote address, when valid, is used as fallback
+// when no IP address has been found at all.
+func ClientIP(hdrs map[string][]string, hasCanonicalMIMEHeaderKeys bool, remoteAddr string, monitoredHeaders []string) (remoteIP, clientIP netip.Addr) {
```
[the remark this answers is not in the document]
Comment (Julio-Guerra) [reply httpsec/client_ip.go]: Renamed in [d1d43fb](https://github.com/DataDog/appsec-internal-go/pull/5/commits/d1d43fb0cf134223967c685700e49532fc867674)

File: httpsec/client_ip.go [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+func ClientIPTags(remoteIP, clientIP netip.Addr) (tags map[string]string) {
+	remoteIPValid := remoteIP.IsValid()
+	clientIPValid := clientIP.IsValid()
+	if !remoteIPValid && !clientIPValid {
+		return nil
+	}
+
+	tags = make(map[string]string, 2)
+	if remoteIPValid {
+		tags["network.client.ip"] = remoteIP.String()
```
Comment (Hellzy) [inline httpsec/client_ip.go:32]: We should probably use `RemoteIPTag` here
```suggestion
		tags[RemoteIPTag] = remoteIP.String()
```

File: httpsec/client_ip.go [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+	}
+	if clientIPValid {
+		tags["http.client_ip"] = clientIP.String()
```
Comment (Hellzy) [inline httpsec/client_ip.go:35]: Same with `ClientIPTag`
```suggestion
		tags[ClientIPTag] = clientIP.String()
```

**Review by Hellzy — CHANGES_REQUESTED** [review]

**Review by Hellzy — APPROVED** [review]

**Pull request merged**